src/_pytest/warning_types.py

```diff
@@ -48,14 +48,13 @@ class PytestDeprecationWarning(PytestWarning, DeprecationWarning):
     __module__ = "pytest"
 
 
-@final
 class PytestRemovedIn8Warning(PytestDeprecationWarning):
     """Warning class for features that will be removed in pytest 8."""
 
     __module__ = "pytest"
 
 
-class PytestReturnNotNoneWarning(PytestDeprecationWarning):
+class PytestReturnNotNoneWarning(PytestRemovedIn8Warning):
```

Comment:
Sorry, I don't recall the original discussion, do we ever plan to make returning non-None an error? Do you have a link to the original discussion.

Reply:
https://github.com/pytest-dev/pytest/pull/9956#issuecomment-1132710934

Reply:
Ahh thanks!